src/map/utils/battleutils.cpp

```diff
@@ -2696,8 +2684,7 @@ namespace battleutils
 		{
 			CCharEntity* PChar = (CCharEntity*)PDefender;
 
-			if (PDefender->GetMJob() == 9 && PDefender->GetMLevel() >= 75)	//Don't want to make a gorillion hasTrait checks
-				MeritBonus = PChar->PMeritPoints->GetMeritValue(MERIT_KILLER_EFFECTS, (CCharEntity*)PDefender);
+			MeritBonus = PChar->PMeritPoints->GetMeritValue(MERIT_KILLER_EFFECTS, (CCharEntity*)PDefender);
```

Comment:
i realise you didnt write this but
```
auto PChar = dynamic_cast<CCharEntity*>(PDefender);

if (PChar)
{
    MeritBonus = PChar->PMeritPoints->GetMeritValue(MERIT_KILLER_EFFECTS, PChar);
}
instead of c style casting PDefender to PChar again anyways